### PR TITLE
[ALLUXIO-2912] Separate KeyValueMaster.java into an interface class and an implementation class

### PR DESCRIPTION
**Pull request by maboelhassan**
https://alluxio.atlassian.net/browse/ALLUXIO-2912

### CODE REVIEW

File: keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterFactory.java [hunk 1]
```diff
@@ -49,13 +49,14 @@ public String getName() {
   }
 
   @Override
-  public KeyValueMaster create(MasterRegistry registry, JournalFactory journalFactory) {
+  public DefaultKeyValueMaster create(MasterRegistry registry, JournalFactory journalFactory) {
```
Comment (apc999) [inline keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterFactory.java]: This should still be `KeyValueMaster`

File: keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterFactory.java [hunk 2]
```diff
@@ -49,13 +49,14 @@ public String getName() {
...
     Preconditions.checkArgument(journalFactory != null, "journal factory may not be null");
     LOG.info("Creating {} ", KeyValueMaster.class.getName());
     Journal journal = journalFactory.create(getName());
     FileSystemMaster fileSystemMaster = registry.get(FileSystemMaster.class);
-    KeyValueMaster keyValueMaster = new KeyValueMaster(fileSystemMaster, journal);
-    registry.add(KeyValueMaster.class, keyValueMaster);
-    return keyValueMaster;
+    DefaultKeyValueMaster defaultKeyValueMaster =
+        new DefaultKeyValueMaster(fileSystemMaster, journal);
+    registry.add(DefaultKeyValueMaster.class, defaultKeyValueMaster);
```
Comment (apc999) [inline keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterFactory.java]: `registry.add(KeyValueMaster.class, defaultKeyValueMaster);`

**Comment by apc999** [comment]
@maboelhassan thanks for the contribution!
I left two comments. Otherwise it looks good to me

**Comment by maboelhassan** [comment]
@apc999 Thanks :)
Applied your feedback in the latest changes.

**Comment by apc999** [comment]
thanks for the contribution! @maboelhassan

**Pull request merged**